lib/galaxy/jobs/runners/arc.py

```diff
@@ -0,0 +1,593 @@
+
+            dataset_name = job_file['name']
+            dataset_id = job_file['dataset_id']
+            dataset_path = job_inputfiles_galaxy_paths[dataset_id]
+            isexe = job_file['isexe']
+           
+            """ Populate the arcjob object with the source - pyarcrest expects this"""
+            arcjob.inputFiles[dataset_name] = 'file://' + dataset_path
+
+            """ Datastaging tag """
+            sub_el = SubElement(datastaging,"InputFile")
+            subsub_el = SubElement(sub_el,"Name")
+            subsub_el.text = dataset_name
+
+            if isexe:
+                """ Fill the appropriate job description fields expected for executables"""
+                """ App tag """
+                sub_el = SubElement(app_exe,"Path")
+                sub_el.text = "./" + dataset_name
+
+
+        """ Populate datastaging output tag with all output files - in addition to populate the arcjob object"""
+        """ Potentially more than one file - but currently actually only one, so the for-loop here is currently not actually needed """
+ 
+        """ Fill the appropriate job description fields """
+        sub_el = SubElement(datastaging,"OutputFile")
+        subsub_el = SubElement(sub_el,"Name")
+        subsub_el.text = "/"
+        
+
+        """ This is hardcoded stdout and stderr files from the ARC job defined here in the runner - TODO - not hardcoded """
+        sub_el = SubElement(datastaging,"OutputFile")
+        subsub_el = SubElement(sub_el,"Name")
+        subsub_el.text = std_out
+
+        """ This is hardcoded stdout and stderr files from the ARC job defined here in the runner - TODO - not hardcoded """
+        sub_el = SubElement(datastaging,"OutputFile")
+        subsub_el = SubElement(sub_el,"Name")
+        subsub_el.text = std_err
+
+        """ TODO - just a sample, this will probably be set by the destination itself - to be discussed """
+        cpuhrs = arc_cpuhrs
+        sub_el = SubElement(resources,"IndividualCPUTime")
+        sub_el.text = arc_cpuhrs
+
+        """ TODO - just a sample, this will probably be set by the destination itself - to be discussed """
+        mem = arc_mem
+        sub_el = SubElement(resources,"IndividualPhysicalMemory")
+        sub_el.text = arc_mem
+
+        """ Populate the arcjob object with rest of necessary and useful fields including the full job description string"""
+        """ All files that should be collected by ARC when the job is finished need to be appended to the downloadFiles list - 
+        here it is just the folder / and all files in the folder will be downloaded. 
+        The arc.py in pyarcrest loops over this list to fetch all outputfiles  """
+        arcjob.downloadFiles.append("/")
+        arcjob.StdOut = std_out
+        arcjob.StdErr = std_err
+        arcjob.RequestedTotalCPUTime = cpuhrs
+        arcjob.descstr = tostring(descr, encoding='unicode',method='xml')
+
+
+    def job_actions(self, arcid, action):
+
```

Comment:
Also - can you just make this two separate methods with two separate returns. This sort method dispatching weakens our tool chains ability to do static checking and I think makes the code a little bit harder to read.

Reply:
Thanks, will look at it!